### ISSUE TITLE
Importing Agda.Builtin.Size in one module affects another module

### ISSUE THREAD

**Issue by nad** [agda/agda#3694]
`Bug2.agda`:
```agda
module Bug2 where

postulate
  F : (A : Set) (B : A → Set) → ((x : A) → B x) → Set

variable
  A : Set
  P : A → Set

postulate
  f : (r : (x : _) → P x) → F _ P (λ x → r x)
```
`Bug.agda`:
```agda
{-# OPTIONS --sized-types #-}

module Bug where

import Agda.Builtin.Size

import Bug2
```
Agda accepts the code in `Bug2.agda`:
```
$ agda --ignore-interfaces Bug2.agda
Checking Bug2 ([…]/Bug2.agda).
```
However, if `Bug2.agda` is loaded via `Bug.agda`, then Agda complains:
```
$ agda --ignore-interfaces Bug.agda
Checking Bug ([…]/Bug.agda).
 Checking Bug2 ([…]/Bug2.agda).
[…]/Bug.agda:7,1-12
Failed to solve the following constraints:
  Is not empty type of sizes: _P.A_16
when scope checking the declaration
  import Bug2
[…]/Bug.agda:7,1-12
Unsolved metas at the following locations:
  […]/Bug2.agda:11,22-23
when scope checking the declaration
  import Bug2
```
Importing `Agda.Builtin.Size` in `Bug.agda` should not affect `Bug2.agda`.

**Comment by andreasabel** [comment]
This seems to be a problem with the constraint machinery, probably in connection to the new generalization feature.
The constraint `Is not empty type of sizes: _P.A_16` is stuck on a meta which comes from generalization.  Likely, this variable was not abstracted over _because_ of the constraint.  However, this particular constraint should _not_ prevent generalization, since generalization will satisfy it.

**Comment by andreasabel** [comment]
@UlfNorell I stalled my attempts to fix this problem since the present solution raises fundamental questions on its correctness: https://github.com/agda/agda/commit/514ca28126740353d7f16de38ce3b691aed81985#r33170209

**Comment by nad** [comment]
> This seems to be a problem with the constraint machinery, probably in connection to the new generalization feature.

Are you suggesting that the following code should be accepted?
```agda
{-# OPTIONS --sized-types #-}

import Agda.Builtin.Size

postulate
  F : (A : Set) (B : A → Set) → ((x : A) → B x) → Set

variable
  A : Set
  P : A → Set

postulate
  f : (r : (x : _) → P x) → F _ P (λ x → r x)
```
Perhaps it should, but I think it is even more serious that the result of checking `Bug2.agda` depends on how it is imported.

**Comment by gallais** [comment]
I get the same error in https://github.com/gallais/generic-syntax (for instance [this build on travis](https://travis-ci.org/gallais/generic-syntax/builds/524556150#L1314))
every time the module is checked for the first time. The second import
goes through without any problem.

**Comment by UlfNorell** [comment]
I fixed the problem with generalizable variables and filed a new issue for the more general issue with size builtins.

**Comment by asr** [comment]
> I fixed the problem with generalizable variables

Please cherry-pick your fix into `release-2.6.0.1`.

> and filed a new issue for the more general issue with size builtins.

Which is the new issue?

**Comment by UlfNorell** [comment]
Cherry picked. The new issue is referenced just above (#3738).
